scripts/screeps_rl_training_runner.py

```
@@ -966,6 +967,96 @@ def scale_validation_environment_containers(variant: JsonObject) -> list[JsonObj
+                "candidateParameterScope": "metadata_only",
+                "reason": "variant summary did not include runtime parameter injection evidence",
+            })
+    observed_ids = {text_or_none(row.get("variantId")) for row in rows}
+    for missing in sorted(expected_ids - observed_ids):
+        rows.append({
+            "variantId": missing,
+            "status": "missing",
+            "runtimeParameterInjection": False,
+            "candidateParameterScope": "metadata_only",
+            "reason": "variant was missing from training results",
+        })
```

Ruff (0.15.13)

[warning] Use `list.extend` to create a transformed list

(PERF401)